### PR TITLE
Minor solver changes

### PR DESCRIPTION
**Pull request by Naktakala**

### CODE REVIEW

File: framework/math/Functions/function_dimA_to_dimB.cc [hunk 1]
```diff
@@ -18,12 +20,46 @@ chi::InputParameters FunctionDimAToDimB::GetInputParameters()
...
+  ChiLogicalError("No available function");
+}
+double FunctionDimAToDimB::ScalarFunctionSlope1Parameter(double) const
+{
+  ChiLogicalError("No available function");
+}
+double FunctionDimAToDimB::ScalarFunctionCurvature1Parameter(double) const
 {
+  ChiLogicalError("No available function");
 }
```
Comment (Naktakala) [inline framework/math/Functions/function_dimA_to_dimB.cc:42]: Don't mind these functions too much. All of this still needs to be optimized

File: framework/physics/SolverBase/chi_solver.cc [hunk 2]
```diff
@@ -101,6 +103,8 @@ Solver::InitTimeStepper(const chi::InputParameters& params)
       custom_params.AddParameter(params.GetParam("max_time_steps"));
     }
 
+    valid_params.AssignParameters(custom_params);
```
Comment (Naktakala) [inline framework/physics/SolverBase/chi_solver.cc:106]: This was actually a bug

File: framework/physics/SolverBase/chi_solver.cc [hunk 3]
```diff
@@ -176,4 +180,37 @@ Solver::GetInfoWithPreCheck(const chi::ParameterBlock& params) const
...
+      timestepper_->SetTime(param.GetValue<double>());
+    if (param_name == "start_time")
+      timestepper_->SetStartTime(param.GetValue<double>());
+    if (param_name == "end_time")
+      timestepper_->SetEndTime(param.GetValue<double>());
+    if (param_name == "max_time_steps")
+      timestepper_->SetMaxTimeSteps(param.GetValue<int>());
+    if (param_name == "dt_min")
+      timestepper_->SetMinimumTimeStepSize(param.GetValue<int>());
+  }
```
Comment (Naktakala) [inline framework/physics/SolverBase/chi_solver.cc:213]: The above is a pretty nice paradigm. Notice the changes in the PRK solver just for me to see if we can link the documentation together.

File: framework/physics/SolverBase/lua/physics_solver_lua_utils.cc [hunk 4]
```diff
@@ -10,9 +10,6 @@
 #include "chi_log.h"
```
Comment (Naktakala) [inline framework/physics/SolverBase/lua/physics_solver_lua_utils.cc:10]: In this file I just renamed LuaSolver to doc_PhysicsSolver

File: framework/post_processors/PostProcessor.cc [hunk 5]
```diff
@@ -128,7 +128,8 @@ void PostProcessor::PushOntoStack(std::shared_ptr<ChiObject>& new_object)
   auto new_subscriber = std::dynamic_pointer_cast<chi::EventSubscriber>(pp_ptr);
 
   ChiLogicalErrorIf(
-    not pp_ptr, "Failure to cast chi::PostProcessor to chi::EventSubscriber");
+    not new_subscriber,
+    "Failure to cast chi::PostProcessor to chi::EventSubscriber");
```
Comment (Naktakala) [inline framework/post_processors/PostProcessor.cc:133]: This was a bug

File: modules/PointReactorKinetics/point_reactor_kinetics.cc [hunk 6]
```diff
@@ -278,4 +278,28 @@ std::vector<double> TransientSolver::SolutionNew() const
...
+{
+  chi_physics::Solver::SetProperties(params);
+
+  for (const auto& param : params)
+  {
+    const std::string& param_name = param.Name();
+    if (param_name == "rho") SetRho(param.GetValue<double>());
+  }
+}
+
```
Comment (Naktakala) [inline modules/PointReactorKinetics/point_reactor_kinetics.cc:304]: This was just me testing what the documentation looks like. Its pretty nice:
image

**Review by zachhardy — APPROVED** [review]
Doesn't touch anything critical.

Is this documentation stuff a new way to do documentation? Could this not be achieved with the standard way we have been doing it? 

As a future work not, FunctionDimAToDimB doesn't have a clear meaning to me. I think it means a function that takes a scalar (A=1) and returns a vector (B=N), or vice versa, but am not sure. Documentation would be nice.

**Comment by Naktakala** [comment]
> Doesn't touch anything critical.
> 
> Is this documentation stuff a new way to do documentation? Could this not be achieved with the standard way we have been doing it?
> 
> As a future work not, FunctionDimAToDimB doesn't have a clear meaning to me. I think it means a function that takes a scalar (A=1) and returns a vector (B=N), or vice versa, but am not sure. Documentation would be nice.

It's a little tricky documenting it as single function that might have many different parameters. But I will think about a way.

Functions: Yes! Unsure if that will stay n-dimensional to n-dimensional.

**Pull request merged**